File: jjx-server/src/main/java/com/jjx/common/enums/StatusEnum.java
package com.jjx.common.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 通用状态枚举
 */
@Getter
@AllArgsConstructor
public enum StatusEnum {

    /** 正常 */
    NORMAL(1, "正常"),
    /** 停用 */
    DISABLE(0, "停用"),
    /** 删除 */
    DELETED(2, "删除");

    private final Integer code;
    private final String info;

    /**
     * 根据code获取枚举
     */
    public static StatusEnum valueOfCode(Integer code) {
        for (StatusEnum status : values()) {
            if (status.getCode().equals(code)) {
                return status;
            }
        }
        return null;
    }

    /**
     * 判断code是否有效
     */
    public static boolean isValid(Integer code) {
        return valueOfCode(code) != null;
    }
}

File: jjx-web/src/enums/common/StatusEnum.ts
// src/enums/common/StatusEnum.ts
import { createNamedEnum } from '../base'

/**
 * 通用状态枚举
 * 对应后端 common StatusEnum（NORMAL=1正常 / DISABLE=0停用 / DELETED=2删除）
 * 2026-09-05（任务1426）：createEnum 列表式改为 createNamedEnum 具名式，供网版启停等通用状态比较使用
 */
export const CommonStatusEnum = createNamedEnum(
  {
    NORMAL: { value: 1, label: '正常', tagProps: { type: 'success' } },
    DISABLED: { value: 0, label: '停用', tagProps: { type: 'danger' } },
    DELETED: { value: 2, label: '删除', tagProps: { type: 'info' } },
  },
  { type: 'info' }
)
